### PR TITLE
Doc update working example

### PR DESCRIPTION
**Pull request by SrikanthPagadarai**

### CODE REVIEW

File: doc/working_example.md [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+The workflow of Genalyzer is best understood with the help of a working example. We use Python to illustrate this workflow in the current section. The data that we will process using Genalyzer for this example is also generated using Genalyzer. It consists of a ``300 KHz`` tone sampled at ``3 MSPS``. The only source of noise in this synthetic data is quantization noise.
+<br/><br/>
+
+### Compute FFT
+We begin with the initial steps involved, _i.e.,_, configuring Genalyzer and performing FFT on the generated data. Since spectral analysis is the core of Genalyzer, computing FFT is the first important step. Expand the code-snippet below for details.
+<details>
+  <summary>Code Snippet</summary>
+
+``` Python
+def main():
```
Comment (tfcollins) [inline doc/working_example.md:12]: Remove use of the main function abstraction. It just makes the examples confusing and wastes space

Comment (SrikanthPagadarai) [reply doc/working_example.md]: Updated.

**Comment by tfcollins** [comment]
myst is complaining about cross references. 
```bash
WARNING: 'myst' cross-reference target not found
```

Need to add this to the conf.py
```diff
# -- General configuration ---------------------------------------------------

# Add any Sphinx extension module names here, as strings. They can be
# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
# ones.
extensions = ["breathe", "sphinx.ext.autodoc", "myst_parser", "sphinx_inline_tabs", "sphinx.ext.graphviz"]

+# Fix use of autogenerated cross-reference warning
+myst_heading_anchors = 2

# Add any paths that contain templates here, relative to this directory.
templates_path = ["_templates"]

```

**Comment by SrikanthPagadarai** [comment]
> myst is complaining about cross references.
> 
> ```shell
> WARNING: 'myst' cross-reference target not found
> ```
> 
> Need to add this to the conf.py
> 
> ```diff
> # -- General configuration ---------------------------------------------------
> 
> # Add any Sphinx extension module names here, as strings. They can be
> # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
> # ones.
> extensions = ["breathe", "sphinx.ext.autodoc", "myst_parser", "sphinx_inline_tabs", "sphinx.ext.graphviz"]
> 
> +# Fix use of autogenerated cross-reference warning
> +myst_heading_anchors = 2
> 
> # Add any paths that contain templates here, relative to this directory.
> templates_path = ["_templates"]
> ```

Updated conf.py.

File: doc/working_example.md [hunk 2]
```diff
@@ -0,0 +1,825 @@
+# Genalyzer Through Examples
+## A Minimal Working Example
+The workflow of Genalyzer is best understood with the help of a working example. We use Python to illustrate this workflow in the current section. The data that we will process using Genalyzer for this example is also generated using Genalyzer. It consists of a ``300 KHz`` tone sampled at ``3 MSPS``. The only source of noise in this synthetic data is quantization noise.
```
Comment (tfcollins) [inline doc/working_example.md:3]: >  We use Python to illustrate this workflow in the current section. The data that we will process using Genalyzer for this example is also generated using Genalyzer. It consists of a ``300 KHz`` tone sampled at ``3 MSPS``. The only source of noise in this synthetic data is quantization noise.

This doesn't flow very well. Can you simplify it a bit?

File: doc/working_example.md [hunk 3]
```diff
@@ -0,0 +1,825 @@
...
+<br/><br/>
+
+### Compute FFT
+We begin with the initial steps involved, _i.e.,_, configuring Genalyzer and performing FFT on the generated data. Since spectral analysis is the core of Genalyzer, computing FFT is the first important step. Expand the code-snippet below for details.
```
Comment (tfcollins) [inline doc/working_example.md:7]: >  Expand the code-snippet below for details.

Remove the text and keep the code expanded by default

File: doc/working_example.md [hunk 4]
```diff
@@ -0,0 +1,825 @@
...
+
+``` Python
+import genalyzer.advanced as gn
+
+#
+# Signal 
+#
+npts = 30000 # number of points in the signal
+freq = 300000 # tone frequency
+phase = 0.0 # tone phase
```
Comment (tfcollins) [inline doc/working_example.md:19]: need units

File: doc/working_example.md [hunk 5]
```diff
@@ -0,0 +1,825 @@
...
+nfft = int(npts/navg) # FFT-order
+window = gn.Window.NO_WINDOW # window function to apply
+
+#
+# Genarate signal for analysis
+#
+awfi = gn.cos(npts, fs, ampl, freq, phase)
+awfq = gn.sin(npts, fs, ampl, freq, phase)
+qwfi = gn.quantize(awfi, fsr, qres, qnoise, code_fmt)
+qwfq = gn.quantize(awfq, fsr, qres, qnoise, code_fmt)
```
Comment (tfcollins) [inline doc/working_example.md:42]: This should go above FFT Config to be near the other signal generation stuff

File: doc/working_example.md [hunk 6]
```diff
@@ -0,0 +1,825 @@
...
+
+#
+# Compute FFT
+#
+fft_cplx = gn.fft(qwfi, qwfq, qres, navg, nfft, window, code_fmt)
+```
+</details>
+
+We will expand this code snippet to illustrate the various capabilities of Genalyzer. We postpone the discussion on how to generate synthetic data using Genalyzer and the related knobs available. We begin first with the FFT-computing step in the above snippet.
```
Comment (tfcollins) [inline doc/working_example.md:51]: >We will expand this code snippet to illustrate the various capabilities of Genalyzer. We postpone the discussion on how to generate synthetic data using Genalyzer and the related knobs available. We begin first with the FFT-computing step in the above snippet.

Remove this text as it doesn't add anything

File: doc/working_example.md [hunk 7]
```diff
@@ -0,0 +1,825 @@
...
+
+The arguments that ``fft()`` takes are shown on [here](#genalyzer.advanced.advanced.fft). As shown on this page, it takes a variable number of arguments to handle several usecases. The usecases are split based on how the I/Q samples are represented, _i.e._, whether the samples are complex-valued, or interleaved I/Q, or split into I and Q separately. These categories are further split based on whether the samples are represented as floating-point or fixed-point integer data. In the latter case, the resolution in bits and code-format are the required inputs. The supported datatypes for the input data vector are: ``complex128``, ``float64``, ``int16``, ``int32``, ``int64``.
```
Comment (tfcollins) [inline doc/working_example.md:53]: Refrain from using "i.e." a lot. It can be distracting.

I don't really understand the point of this paragraph. Are you trying to explain how to use the fft function? its flexibility? or its nuanced API? 

For individual arguments just refer users to the reference doc. This shouldn't be a copy+paste of that here. 

This section is "Compute FFT" but it probably needs to be more specific on the data types used since that's what drives everything.

File: doc/working_example.md [hunk 8]
```diff
@@ -0,0 +1,825 @@
...
+signal_tone_key = 'A' # signal-key
+ssb_fund = 0 # number of single-side bins
+gn.fa_create(object_key)
+gn.fa_fsample(object_key, fs)
+gn.fa_max_tone(object_key, signal_tone_key, gn.FaCompTag.SIGNAL, ssb_fund)
+```
+</details>
+
+##### Create ``fourier_analysis`` Object
+The first step in configuring Genalyzer is to call the ``fa_create()`` function with an _object\_key_. This key is used for all configuring Genalyzer further, and for computing and retrieving the metrics.
```
Comment (tfcollins) [inline doc/working_example.md:129]: Aren't we on like step 2 at this point?

>This key is used for all configuring Genalyzer further

I think you meant configuration and not configuring

Can we use some other name than "object_key"? Is there something more relatable we can use?

File: doc/working_example.md [hunk 9]
```diff
@@ -0,0 +1,825 @@
...
+```{admonition} Aside
+:class: note
+
+Under the hood, Genalyzer adds a key-value pair to a  ``static`` ``map`` container to manage the metrics to be computed. The key is the string argument passed through ``fa_create()``, and the mapped value is a shared-pointer to an instance of ``fourier_analysis`` class. This key is then used to further configure Genalyzer, and to compute and retrieve the metrics through ``fourier_analysis`` class. The intent behind using a ``map`` container is to be easily able to associate multiple keys to different snapshots of the data being analyzed and to have the metrics for each of those snapshots available.
```
Comment (tfcollins) [inline doc/working_example.md:133]: Is there a way to inspect it?

File: doc/working_example.md [hunk 10]
```diff
@@ -0,0 +1,825 @@
...
+```
+
+##### Set Sample-Rate
+Set the sample-rate of the data vector using ``fa_fsample``. As shown on this [page](#genalyzer.advanced.advanced.fa_fsample), we associate this configuration step with an _object\_key_.
+<br/><br/>
+
+##### Identify Signal Tone
+The next step is to identify the signal tone and label it. This is done by calling either [``fa_fixed_tone()``](#genalyzer.advanced.advanced.fa_fixed_tone) or [``fa_max_tone()``](#genalyzer.advanced.advanced.fa_max_tone) functions. In both cases, the first two arguments are _keys_. The first _key_ is the _object\_key_ mentioned previously, and the second _key_ is a string that is to be associated with the signal component whose magnitude is measured and used subsequently when computing performance metrics. In both cases, the third argument is a tag which maps to the component type. To identify a component as the signal tone, we use ``GnFACompTagSignal``. The other tags available are shown [here](#genalyzer.advanced.advanced.FaCompTag). As the name indicates, ``fa_max_tone()`` interprets the tone with the highest magnitude as the signal tone.
```
Comment (tfcollins) [inline doc/working_example.md:141]: >The next step is to identify the signal tone and label it. This is done by calling either [``fa_fixed_tone()``](#genalyzer.advanced.advanced.fa_fixed_tone) or [``fa_max_tone()``](#genalyzer.advanced.advanced.fa_max_tone) functions. In both cases, the first two arguments are _keys_. The first _key_ is the _object\_key_ mentioned previously, and the second _key_ is a string that is to be associated with the signal component whose magnitude is measured and used subsequently when computing performance metrics. In both cases, the third argument is a tag which maps to the component type. To identify a component as the signal tone, we use ``GnFACompTagSignal``. The other tags available are shown [here](#genalyzer.advanced.advanced.FaCompTag). As the name indicates, ``fa_max_tone()`` interprets the tone with the highest magnitude as the signal to

The descriptions of keys and tags here is a bit confusing. Is there a plain way to describe what they are for?

File: doc/working_example.md [hunk 11]
```diff
@@ -0,0 +1,825 @@
...
+
+```{note}
+Since ``fa_fixed_tone()`` requires that the frequency of a desired tone is provided as an argument, it is useful for taking into account any desired tone that is pertinent to analysis.
```
Comment (tfcollins) [inline doc/working_example.md:144]: > it is useful for taking into account any desired tone that is pertinent to analysis.

I think its more than useful, its actually required in this case right?

**Review by tfcollins — CHANGES_REQUESTED** [review]

File: bindings/python/fft.png
Comment (tfcollins) [inline bindings/python/fft.png:1]: Image files should not be in source code folders. They need to be located with doc.

File: doc/spectral_analysis.md
Comment (tfcollins) [inline doc/spectral_analysis.md:1]: This is no real code in the page. Am I missing something?

**Review by tfcollins — CHANGES_REQUESTED** [review]
See comments

File: bindings/python/fft.png
Comment (SrikanthPagadarai) [reply bindings/python/fft.png:1]: Fixed.

File: doc/spectral_analysis.md
Comment (SrikanthPagadarai) [reply doc/spectral_analysis.md:1]: I'm pointing readers to files in this manner: 
https://github.com/analogdevicesinc/genalyzer/blob/31b210be3da07f116984752344640ac1fb46d453/doc/spectral_analysis.md?plain=1#L164

Comment (tfcollins) [reply doc/spectral_analysis.md:1]: This is really hard to follow. Code snippets should be in the doc so people don't have to keep context switching

**Review by tfcollins — CHANGES_REQUESTED** [review]
I think this is in good shape. Can you clean up the git history and we'll merge
